extensions/kill-process/src/index.tsx

```diff
@@ -14,43 +15,61 @@ export default function ProcessList() {
   const shouldShowPID = preferences.shouldShowPID ?? false;
   const shouldShowPath = preferences.shouldShowPath ?? false;
   const refreshDuration = +preferences.refreshDuration;
+  const [sortByMem, setSortByMem] = useState<boolean>(preferences.sortByMem ?? false);
+  const [aggregateApps, setAggregateApps] = useState<boolean>(preferences.aggregateApps ?? false);
 
   const fetchProcesses = () => {
-    exec(`ps -eo pid,pcpu,comm | sort -nrk 2,3`, (err, stdout) => {
+    exec(`ps -eo pid,ppid,pcpu,rss,comm`, async (err, stdout) => {
       if (err != null) {
         return;
       }
 
-      const processes = stdout
+      let processes = stdout
         .split("\n")
         .map((line) => {
-          const [, id, cpu, path] = line.match(/(\d+)\s+(\d+[.|,]\d+)\s+(.*)/) ?? ["", "", "", ""];
-          const name = path.match(/[^/]*[^/]*$/i)?.[0] ?? "";
+          const defaultValue = ["", "", "", "", "", ""];
+          const regex = /(\d+)\s+(\d+)\s+(\d+[.|,]\d+)\s+(\d+)\s+(.*)/;
+          const [, id, pid, cpu, mem, path] = line.match(regex) ?? defaultValue;
+          const processName = path.match(/[^/]*[^/]*$/i)?.[0] ?? "";
           const isPrefPane = path.includes(".prefPane");
-          const isApp = path.includes(".app");
+          const isApp = path.includes(".app/");
 
           return {
-            id,
-            cpu,
-            path,
-            name,
+            id: parseInt(id),
+            pid: parseInt(pid),
+            cpu: parseFloat(cpu),
+            mem: parseInt(mem),
             type: isPrefPane ? "prefPane" : isApp ? "app" : "binary",
+            path,
+            processName,
           } as Process;
         })
-        .filter((process) => process.name !== "");
+        .filter((process) => process.processName !== "");
+
+      if (aggregateApps) {
+        processes = aggregate(processes);
+      }
+
+      processes.sort((a, b) => {
+        if (sortByMem) {
+          return a.mem > b.mem ? -1 : 1;
+        } else {
+          return a.cpu > b.cpu ? -1 : 1;
+        }
+      });
 
       setState(processes);
     });
   };
 
-  useInterval(fetchProcesses, refreshDuration);
+  useInterval(fetchProcesses, refreshDuration, [sortByMem, aggregateApps]);
```

Comment:
I don't think this kind of aggregate operation need to trigger refresh.

I mean it could aggregate based on the current result only, while the current change is updated, we just aggregate again.

Reply:
It can be done in this way, but I think it's unnecessary to remember the old value, since aggregating and sortByMem is not perform too often, and we can get the latest result, and refreshing seems not cost much time. @rolandleth what do you think?

Reply:
I don't think we need keep old value, I think we just need one more useMemo/useEffect like

```
useEffect( aggregate Fn, [ execResult ])
```

this execResult is only  a reference, whenever this ref changed, it will trigger aggregate Fn.

Reply:
Got it, I'll have a look in it.

Reply:
I'm not sure what do you mean by `useEffect( aggregateFn, [ execResult ])`, what is the `execResult`? The `aggregateFn` is irreversible, without keeping the old value, how do we dis-aggregate by the already aggregated value?

Reply:
First, in the first, usePromise / useCachePromise (exec( ps ...)), will return an execResult, which is basically pure list of process.

Then, if aggregation flag is true, aggregation is performed and render aggregated results, if not, render the list of process.

I think you can consider execResult as "the old value".

My main concern is that

```
  useEffect(() => {
    savedCallback.current();
    const refreshEnabled = (delay ?? 0) > 0;
    if (refreshEnabled) {
      const interval = Math.max(delay ?? 0, 1000);
      const id = setInterval(() => savedCallback.current(), interval);
      return () => clearInterval(id);
    }
  }, [delay]);

```

as you can see, previously, the only deps in delay, which won't be changed per command invocation, which means this useEffect is triggered once, and this useEffect is to create a scheduled refresh. 

If more deps are introduced, more scheduled process will be introduced as well, and most importantly, which is not necessary.

Reply:
I want to make sure that you aware that `aggregateApps` may changed during per command invocation by the action menu? If I use `useEffect( aggregateFn, [ execResult ])` and `aggregateApps` changed, the `state` won't update. If you mean `useEffect( aggregateFnIfNeeded, [setIntervalResult, aggregateApps, sortByMem])`, then I have to store both `setIntervalResult` and the result in aggregateFnIfNeeded.

If I introduce deps to `useInterval`, I think there will be only one interval loop at same time because I've tested it, there will still refresh every 3 seconds after I changed the `aggregateApps` or `sortByMem` value, and in my understanding, the `() => clearInterval(id);` will be executed after changes in [delay, aggregateApps, sortByMem] happened, which will clear last interval loop.

Reply:
I know it is triggered by action.

```

const execResult;
const isAggregateEnableld = false

useInterval((), [delay]);

const aggregateResult = useMemo ( aggregateFn, [execResult, isAggregateEnableld])
```

Reply:
> then I have to store both setIntervalResult and the result in aggregateFnIfNeeded.

Yes I thought there were some way not storing two value

Reply:
Resolved by adding `fetchResult` property to not refresh on change of sortByMem or aggregateApps